### PR TITLE
Add S3 AWS Plugin to auto copy output files to S3

### PR DESCRIPTION
**Pull request by Vorobeyko**

### CODE REVIEW

**Comment by DavertMik** [comment]
Nice! 
However, I would recommend to remove Amazon S3 from direct dependencies. And ask user to install that manually. Check for that dependency on start, and throw exception asking user to install it.

**Comment by Vorobeyko** [comment]
> Nice!
> However, I would recommend to remove Amazon S3 from direct dependencies. And ask user to install that manually. Check for that dependency on start, and throw exception asking user to install it.

OK!

**Comment by Vorobeyko** [comment]
@DavertMik done =)

File: package.json [hunk 1]
```diff
@@ -53,6 +53,7 @@
     "inquirer": "^6.3.1",
     "js-beautify": "^1.9.1",
     "lodash.merge": "^4.6.1",
+    "mime-types": "^2.1.24",
```
Comment (DavertMik) [inline package.json]: I'd rather not add an extra library for just a plugin.
I see two options here:
1. ask to install mime-types as well as aws-sdk
2. move this plugin into a standalone package (probably the cleanest approach).

I think the option 2 would be the best:
- AWS SDK and mime-types will be installed automatically
- plugin will be separated from the core 

So I think we should consider moving this plugin into `@codeceptjs/s3` package.
We can still add it to official list on the website or to 3rd-party list at https://github.com/Codeception/CodeceptJS/wiki/Community-Helpers

**Comment by Vorobeyko** [comment]
I choose the second option!)

**Comment by DavertMik** [comment]
Moved to https://github.com/Codeception/codeceptjs-plugins

**Pull request closed without merging**